src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -499,13 +500,27 @@ private function handleUseItemTransaction(UseItemTransactionData $data) : bool{
 
 				$blockPos = $data->getBlockPosition();
 				$vBlockPos = new Vector3($blockPos->getX(), $blockPos->getY(), $blockPos->getZ());
-				$this->player->interactBlock($vBlockPos, $data->getFace(), $clickPos);
-				if($data->getClientInteractPrediction() === PredictedResult::SUCCESS){
-					//always sync this in case plugins caused a different result than the client expected
-					//we *could* try to enhance detection of plugin-altered behaviour, but this would require propagating
-					//more information up the stack. For now I think this is good enough.
-					//if only the client would tell us what blocks it thinks changed...
-					$this->syncBlocksNearby($vBlockPos, $data->getFace());
+				if($this->player->canInteract($vBlockPos->add(0.5, 0.5, 0.5), 15)){
```

Comment:
I actually meant to just do a distance check like the one we do in `syncBlocksNearby`. I don't remember what this function does, but it might be more complex than we need. Also, hardcoding a max distance of 15 might get in the way if plugins customise reach distance.